kyo-core/shared/src/main/scala/kyo/envs.scala

```diff
@@ -1,32 +1,36 @@
 package kyo
 
 import kyo.core.*
+import scala.util.NotGiven
 
 class Envs[+V] extends Effect[Envs[V]]:
     type Command[T] = Unit
 
 object Envs:
     private case object envs extends Envs[Any]
-    def apply[V]: Envs[V] = envs.asInstanceOf[Envs[V]]
+    private def envs[V]: Envs[V] = envs.asInstanceOf[Envs[V]]
 
-    extension [V](self: Envs[V])
+    def get[V](using Tag[Envs[V]]): V < Envs[V] =
+        envs[V].suspend[V](())
 
-        def get(using Tag[Envs[V]]): V < Envs[V] =
-            self.suspend[V](())
-
-        inline def use[T, S](inline f: V => T < S)(
+    class UseDsl[V]:
+        inline def apply[T, S](inline f: V => T < S)(
             using inline tag: Tag[Envs[V]]
         ): T < (Envs[V] & S) =
-            self.suspend[V, T, S]((), f)
+            envs[V].suspend[V, T, S]((), f)
+    end UseDsl
 
-        def run[T: Flat, S, VS, VR](env: V)(value: T < (Envs[VS] & S))(
-            using
-            HasEnvs[V, VS] { type Remainder = VR },
-            Tag[Envs[V]]
-        ): T < (S & VR) =
-            Envs[V].handle(handler[V])(env, value).asInstanceOf[T < (S & VR)]
+    def use[V](
+        using ng: NotGiven[V => Nothing]
+    ): UseDsl[V] =
+        new UseDsl[V]
 
-    end extension
+    def run[V, T: Flat, S, VS, VR](env: V)(value: T < (Envs[VS] & S))(
```

Comment:
I suspect we may need to improve manually providing type `V` in the case where `env` needs to be a specific type (think type that extends multiple traits) - though don't have any way to confirm that. Perhaps worth proactively adding a DSL.

Reply:
Minor: I would rename `env` -> `value` and `value` -> `effect`

Reply:
yeah, I think we should follow up trying to make parameter naming more explicit and regular in all APIs. I'll keep as is since other APIs typically use `v` or `value` for a computation parameter. I'm not a big fan of `effect`  because in Kyo's context I'd expect an `effect` parameter to be a algebraic effect, not a computation.

Reply:
Makes sense, thanks.